hail/src/main/scala/is/hail/annotations/OrderedRVIterator.scala

```diff
@@ -6,6 +6,21 @@ import is.hail.utils._
 import scala.collection.generic.Growable
 import scala.collection.mutable
 
+object OrderedRVIterator {
+  def multiZipJoin(
+    its: IndexedSeq[OrderedRVIterator]
+  ): Iterator[Array[RegionValue]] = {
+    require(its.length > 0)
+    val first = its(0)
+    val flipbooks = its.map(_.iterator.toFlipbookIterator)
+    FlipbookIterator.multiZipJoin(
+      flipbooks,
+      null,
+      first.t.joinComp(first.t).compare
```

Comment:
`joinComp` does use the unsafe ordering, so creating per-key garbage is not the problem.